packages/asset-swapper/src/utils/market_operation_utils/balancer_utils.ts

```diff
@@ -42,10 +46,52 @@ export class BalancerPoolsCache {
         return Promise.race([this._getPoolsForPairAsync(takerToken, makerToken), timeout]);
     }
 
-    protected async _getPoolsForPairAsync(takerToken: string, makerToken: string): Promise<BalancerPool[]> {
+    public getCachedPoolAddressesForPair(
+        takerToken: string,
+        makerToken: string,
+        cacheExpiryMs?: number,
+    ): string[] | undefined {
+        const key = JSON.stringify([takerToken, makerToken]);
```

Comment:
Would we have two caches for `[DAI,USDC]` and `[USDC,DAI]`? Would sorting the keys help a cache hit?

Reply:
the two aren't exactly interchangeable because we sort the pools by maker token balance. I'm not too concerned about cache misses, bc at least anecdotally (running simbot) the cache warms up pretty quick